aws/resource_aws_media_connect_flow_test.go

```diff
@@ -0,0 +1,290 @@
+					resource.TestCheckResourceAttr(resourceName, "source.0.protocol", "rtp-fec"),
+					resource.TestCheckResourceAttr(resourceName, "source.0.ingest_port", "3333"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSMediaConnectFlowConfig_Decryption(t *testing.T) {
+	var flow mediaconnect.Flow
+
+	rName := fmt.Sprintf("tfacctest%s", acctest.RandString(5))
+	resourceName := "aws_media_connect_flow.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSMediaConnectFlowDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSMediaConnectFlowConfig_Decryption(rName, "aes128"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSMediaConnectFlowExists(resourceName, &flow),
+					testAccMatchResourceAttrRegionalARN(resourceName, "arn", "mediaconnect", regexp.MustCompile(`flow:.+`)),
+					resource.TestCheckResourceAttrSet(resourceName, "egress_ip"),
+					resource.TestCheckResourceAttr(resourceName, "source.#", "1"),
+					testAccMatchResourceAttrRegionalARN(resourceName, "source.0.arn", "mediaconnect", regexp.MustCompile(`source:.+`)),
+					resource.TestCheckResourceAttr(resourceName, "source.0.description", "Managed by Terraform"),
+					resource.TestCheckResourceAttr(resourceName, "source.0.decryption.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "source.0.decryption.0.key_type", "static-key"),
+					resource.TestCheckResourceAttr(resourceName, "source.0.decryption.0.algorithm", "aes128"),
+					resource.TestCheckResourceAttrPair(resourceName, "source.0.decryption.0.role_arn", "aws_iam_role.test", "arn"),
+					resource.TestCheckResourceAttrPair(resourceName, "source.0.decryption.0.secret_arn", "aws_secretsmanager_secret.test", "arn"),
+				),
+			},
+			{
+				Config: testAccAWSMediaConnectFlowConfig_Decryption(rName, "aes256"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSMediaConnectFlowExists(resourceName, &flow),
+					resource.TestCheckResourceAttr(resourceName, "source.0.decryption.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "source.0.decryption.0.algorithm", "aes256"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSMediaConnectFlowConfig_Options(t *testing.T) {
+	var flow mediaconnect.Flow
+
+	rName := fmt.Sprintf("tfacctest%s", acctest.RandString(5))
+	resourceName := "aws_media_connect_flow.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSMediaConnectFlowDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSMediaConnectFlowConfig_Options(rName, "us-west-2a", "Test Description1"),
```

Comment:
Recommend using `aws_availability_zones` data source (and value checking via `resource.TestCheckResourceAttrPair()` instead of hardcoding availability zone information in the testing. 👍

Reply:
Thanks, i fixed to use `aws_availability_zones`  cd34092